src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/PiglinListener.java

```diff
@@ -38,6 +48,30 @@ public void onEntityPickup(EntityPickupItemEvent e) {
         }
     }
 
+    @EventHandler
+    public void onPiglinDropItem(EntityDropItemEvent e) {
+        if (e.getEntity() instanceof Piglin) {
+            Piglin piglin = (Piglin) e.getEntity();
+            Set<ItemStack> drops = SlimefunPlugin.getRegistry().getBarterDrops(); 
+            
+            /*
+             * NOTE:  Getting a new random number each iteration because multiple items could have the same
+             * % chance to drop, and if one fails all items with that number will fail.  Getting a new random number
+             * will allow multiple items with the same % chance to drop.
+             */
+            
+            for (ItemStack is : drops) {
+                SlimefunItem sfi = SlimefunItem.getByItem(is);
+                if (sfi instanceof PiglinBarterDrop && ((PiglinBarterDrop)sfi).getBarteringLootChance() >= ThreadLocalRandom.current().nextInt(100)) {
```

Comment:
You're doing a random number between 0-99 and then a greater OR equal). If I had a 1% chance this would actually be 2% (0 and 1).

Fix pls